### PR TITLE
Add unit tests for DataGridViewTextBoxColumn

### PR DESCRIPTION
**Pull request by Olina-Zhang**
related https://github.com/dotnet/winforms/issues/10453
## Proposed changes

- Add unit tests for DataGridViewTextBoxColumn to test its Constructor, CellTemplate, MaxInputLength, SortMode properties and ToString() method
- Enable nullability in DataGridViewTextBoxColumnTests.cs

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12713)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewTextBoxColumnTests.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    [WinFormsFact]
+    public void DataGridViewTextBoxColumn_CellTemplate_SetInvalidType_ThrowsInvalidCastException()
+    {
+        using DataGridViewButtonCell cell = new();
+
+        Action action = () => _column.CellTemplate = cell;
+        action.Should().Throw<InvalidCastException>()
+            .WithMessage(string.Format(SR.DataGridViewTypeColumn_WrongCellTemplateType, "System.Windows.Forms.DataGridViewTextBoxCell"));
+    }
+
```
Comment (ricardobossan) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewTextBoxColumnTests.cs:53]: Maybe increase code coverage by a few lines with edge cases like this:

```csharp
[Fact]
public static void DataGridViewTextBoxColumn_MaxInputLengthSetNull_DoesNotThrow()
{
    DataGridView? dataGridView = null;
    DataGridViewTextBoxColumn column = new();
    dataGridView?.Columns.Add(column);

    Action action1 = () => column.MaxInputLength = 100;
    action1.Should().NotThrow();
}

[Fact]
public void DataGridViewTextBoxColumn_MaxInputLengthWithNullCellTemplate_ThrowsInvalidOperationException()
{
    _column.CellTemplate = null;
    int maxInput = 0;

    Action action = () => maxInput = _column.MaxInputLength;

    action.Should().Throw<InvalidOperationException>();
}

```

Comment (ricardobossan) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewTextBoxColumnTests.cs:53]: Maybe increase code coverage by a few lines with edge cases like this:

```csharp
[Fact]
public static void DataGridViewTextBoxColumn_MaxInputLengthSetNull_DoesNotThrow()
{
    DataGridView? dataGridView = null;
    DataGridViewTextBoxColumn column = new();
    dataGridView?.Columns.Add(column);

    Action action1 = () => column.MaxInputLength = 100;
    action1.Should().NotThrow();
}

[Fact]
public void DataGridViewTextBoxColumn_MaxInputLengthWithNullCellTemplate_ThrowsInvalidOperationException()
{
    _column.CellTemplate = null;
    int maxInput = 0;

    Action action = () => maxInput = _column.MaxInputLength;

    action.Should().Throw<InvalidOperationException>();
}

```

**Review by ricardobossan — COMMENTED** [review]
Other than a small suggestion, all LGTM!

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewTextBoxColumnTests.cs (hunk 1 above)
Comment (Olina-Zhang) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewTextBoxColumnTests.cs:53]: That's great. Added.

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you!
